src/routes/guides/testing-your-app.mdx

```diff
@@ -0,0 +1,357 @@
+
+ Test Files  [1;32m1 passed[0;8m (1)
+      Tests  [1;32m1 passed[0;8m (1)
+   Start at  [1;1m16:51:19[0;8m
+   Duration  [1;1m4.34s[0;8m (transform 1.01s, setup 205ms, collect 1.54s, tests 155ms,
+environment 880ms, prepare 212ms)
+
+```
+
+### Rendering the component
+
+`render` from `@solidjs/testing-library` sets up a container, renders the component within and registers it for cleanup so that the container will removed after the successful test. It also handles wrapping the component in contexts and set up a router.
+
+```tsx frame="none"
+const renderResult = render(
+  () => <MyComponent />, // @solidjs/testing-library requires a function
+  { // all options are optional
+    container, // manually set up your own container, will not be handled
+    baseElement, // parent of container in case it is not supplied
+    queries, // manually set up custom queries
+    hydrate, // set to `true` to use hydration
+    wrapper, // reusable wrapper component to supply context
+    location, // sets up a router pointed to the location if provided
+  }
+)
+const {
+  asFragment, // function returning the contents of the container
+  baseElement, // the parent of the container
+  container, // the container in which the component is rendered
+  debug, // a function giving some helpful debugging output
+  unmount, // manually removing the component from the container
+  ...queries, // functions to select elements from the container
+} = renderResult
+```
+
+#### Using the right queries
+
+```
+                                          ⎧ Role
+                              get ⎫  By   ⎪ DisplayValue
+                            query ⎬       ⎨ LabelText
+                             find ⎭ AllBy ⎪ Text
+                                          ⎩ ...
+```
+
+The first and second part depends on if the query should wait for the element to (not) appear and if it should throw if the element cannot be found:
+
+- **getBy**: synchronous, throws if not found or more than 1 matches
+- **getAllBy**: synchronous, throws if not found, returns array of matches
+- **queryBy**: synchronous, null if not found, error if more than 1 matches
+- **queryAllBy**: synchronous, returns array of zero or more matches
+- **findBy**: asynchronous, rejected if not found within 1000ms or more than 1 matches, resolves wth element if found
+- **findAllBy**: asynchronous, rejected if not found within 1000ms, resolves with array of one or more element(s)
+
+If the `location` option is used or the component is based on resources, the router will be lazy-loaded, thus the first query after rendering needs to be `find...`. If it should be tested whether something is not rendered, first find something that will be rendered and then use `query...`; for everything else use `get...`. 
+
+The last part depends on the characteristics of the element you want to select. If possible, try to select for accessible attributes (role, label, placeholder, text, ...):
```

Comment:
Actually, that would be a pretty unhelpful rule, as you could obviously query for an input rule in a form with multiple fields, but it would make more sense to go with the label in this case. One should strive for a compromise between having a simple and maintainable query and preferably using only accessible attributes.

Reply:
I've rewrote the section so it matches more with testing library docs.